eclipsePlugin/src/de/tobject/findbugs/decorators/ResourceBugCountDecorator.java

```diff
@@ -34,10 +54,262 @@
+                if (monitor.isCanceled()) {
+                    queue.clear();
+                    return Status.CANCEL_STATUS;
+                } else if (!queue.isEmpty()) {
+                    schedule(100);
+                }
+            }
+            return Status.OK_STATUS;
+        }
+
+        private Entry<BugCountTask, Set<BugCountUpdateJob>> poll() {
+            Entry<BugCountTask, Set<BugCountUpdateJob>> next = null;
+            synchronized (queue) {
+                if (!queue.isEmpty()) {
+                    Iterator<Entry<BugCountTask, Set<BugCountUpdateJob>>> iterator = queue.entrySet().iterator();
+                    next = iterator.next();
+                    iterator.remove();
+                }
+            }
+            return next;
+        }
+
+        void schedule(BugCountTask task, BugCountUpdateJob job) {
+            synchronized (queue) {
+                queue.compute(task, (k, v) -> {
+                    if (v == null) {
+                        v = new LinkedHashSet<>();
+                    }
+                    if (v.add(job)) {
+                        schedule(100);
+                    }
+                    return v;
+                });
+            }
+        }
+    }
+
+    static final class BugCountTask {
+
+        final WorkItem workItem;
+        volatile int oldBugCount;
+        volatile int newBugCount;
+
+        public BugCountTask(WorkItem workItem) {
+            this.workItem = workItem;
+        }
+
+        @Override
+        public int hashCode() {
+            return workItem.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (!(obj instanceof BugCountTask)) {
+                return false;
+            }
+            BugCountTask other = (BugCountTask) obj;
+            return workItem.equals(other.workItem);
+        }
+
+        void run() {
+            IProject project = workItem.getProject();
+            if (project == null) {
+                return;
+            }
+            try {
+                newBugCount = workItem.getMarkerCount(true);
+            } finally {
+                Integer old = BugCountCacheManager.setBugCount(project, newBugCount);
+                if (old != null) {
+                    oldBugCount = old.intValue();
+                }
+            }
+        }
+
+        boolean isBugCountChanged() {
+            return newBugCount != oldBugCount;
+        }
+
+        @Override
+        public String toString() {
+            StringBuilder builder = new StringBuilder();
+            builder.append("BugCountTask ["); //$NON-NLS-1$
+            if (workItem != null) {
+                builder.append("resource="); //$NON-NLS-1$
+                builder.append(workItem);
+                builder.append(", "); //$NON-NLS-1$
+            }
+            builder.append("newBugCount="); //$NON-NLS-1$
```

Comment:
These lines could be fluent.

Reply:
This is just default Eclipse `toString()` generator, and there is no need for any optimization either, just for debugging here.